wagtail/models/__init__.py

```diff
@@ -2074,6 +2095,14 @@ def get_preview_template(self, request, mode_name):
         return self.get_template(request)
 
     def serve(self, request, *args, **kwargs):
+        if request.method not in self.allowed_http_methods_upper:
```

Comment:
Question: Why the property which uppers the methods? This will create new strings for every request, which seems unnecessary.

Django's [`require_http_methods`](https://docs.djangoproject.com/en/5.1/topics/http/decorators/#django.views.decorators.http.require_http_methods) decorator expects methods to be uppercase - I think we can do the same here.

Reply:
Probably because Django's [`View.http_method_names`](https://docs.djangoproject.com/en/stable/ref/class-based-views/base/#django.views.generic.base.View.http_method_names) accept lowercase strings...

Not sure if it _only_ accepts lowercase, though.

Reply:
@laymonage Your deduction is correct. Would be happy to standardise on upper-case everywhere, so that this isn't needed.

Reply:
Is an enum better? Feels like since Django uses strings everywhere, we should follow suit.

Reply:
I feel an enum is better... they are values that already exist in memory, and it's more robust that way... An AttributeError is far more difficult to ignore when you make a typo. It will support uppercase strings too, though

Reply:
I would suspect the reason Django and other places use strings is because HTTPMethod wasn't around until Python 10, and the functionality pre-dates that a fair bit, plus Enums weren't quite as easy to work with at that time.

Reply:
@RealOrangeOne How do you feel about the above rationisation? Are you happy to mark this as resolved?